### PR TITLE
Add GTK4 and QT6 to toolkit testsuite

### PR DESCRIPTION
**Pull request by asdil12**
VR: https://openqa.opensuse.org/tests/3655158

### CODE REVIEW

**Review by foursixnine — APPROVED** [review]
We really need to do something about compressed tarballs...

File: tests/x11/toolkits/gtk4.pm [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
```
Comment (okurz) [inline tests/x11/toolkits/gtk4.pm:3]: For new files you can just leave out the Copyright year. It's wrong anyway :D

File: tests/x11/toolkits/prepare.pm [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # SUSE's openQA tests
 #
-# Copyright 2018 SUSE LLC
+# Copyright 2023 SUSE LLC
 # SPDX-License-Identifier: FSFAP
```
Comment (okurz) [inline tests/x11/toolkits/prepare.pm:4]: Keep the old, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md

File: tests/x11/toolkits/qt6.pm [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
```
Comment (okurz) [inline tests/x11/toolkits/qt6.pm:3]: Delete year

File: tests/x11/toolkits/qt6.pm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    x11_start_program('xterm');
+    assert_script_run 'cd data/toolkits';
+
+    assert_script_run 'make qt6';
+    script_run './qt6', 0;
+    assert_screen 'ui-toolkit-qt6';
+    wait_screen_change { send_key 'alt-f4' };
+    assert_script_run '$(exit $?)';
+
+    enter_cmd "exit";
```
Comment (okurz) [inline tests/x11/toolkits/qt6.pm:26]:
```suggestion
    enter_cmd 'exit';
```

**Review by okurz — CHANGES_REQUESTED** [review]

File: tests/x11/toolkits/gtk4.pm [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    x11_start_program('xterm');
+    assert_script_run 'cd data/toolkits';
+
+    assert_script_run 'make gtk4';
+    script_run './gtk4', 0;
+    assert_screen 'ui-toolkit-gtk4';
+    wait_screen_change { send_key 'alt-f4' };
+
+    assert_script_run '$(exit $?)';
+    enter_cmd "exit";
```
Comment (okurz) [inline tests/x11/toolkits/gtk4.pm:26]:
```suggestion
    enter_cmd 'exit';
```

File: data/toolkits/fltk.cc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#include <FL/Fl.H>
```
Comment (okurz) [inline data/toolkits/fltk.cc:1]: Please add the copyright+license in these source files as well

**Review by okurz — CHANGES_REQUESTED** [review]

**Review by okurz — APPROVED** [review]

**Comment by DimStar77** [comment]
Those two new tests - appearing the first time in TW, are both failing:

https://openqa.opensuse.org/tests/3660785 - qt6 and gtk4 failing

**Comment by DimStar77** [comment]
> VR: https://openqa.opensuse.org/tests/3655158

The VR is two days old and not based on the latest code submission. Even the close of the VR is only 819f12e - quite a few force-pushes happened after that

**Comment by DimStar77** [comment]
> > VR: https://openqa.opensuse.org/tests/3655158
> 
> The VR is two days old and not based on the latest code submission. Even the close of the VR is only [819f12e](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/819f12e4b20b062dad15a18dec78810dc3bbcdf2) - quite a few force-pushes happened after that

The VR is fine - just that the VR was only running on GNOME (where it passes) but was not running on KDE - possibly missing needles (but the screens do not really look like the toolkit's dialogs should look like)

**Comment by asdil12** [comment]
That simple hello world labels/buttons look fine. Just the needles for the theme used on kde were missing.
I added the needles and restarted the latest kde test.

**Pull request merged**